velox/functions/sparksql/tests/SplitTest.cpp

```diff
@@ -0,0 +1,452 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/Udf.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/parse/Expressions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::test;
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SplitTest : public SparkFunctionBaseTest {
+ protected:
+  // Method runs the given split function, f.e. split(C0, C1), where C0 is the
+  // input column and the C1 is delimiter column.
+  // Encoding arguments control what kind of vectors we should create for the
+  // function arguments.
+  // limit should be set to the corresponding limit, if query contains limit
+  // argument (C2).
+  VectorPtr run(
+      const std::vector<std::string>& input,
+      const std::string& delim,
+      const char* query,
+      std::optional<int32_t> limit = std::nullopt,
+      VectorEncoding::Simple encodingStrings = VectorEncoding::Simple::FLAT,
+      VectorEncoding::Simple encodingDelims = VectorEncoding::Simple::CONSTANT,
+      VectorEncoding::Simple encodingLimit = VectorEncoding::Simple::CONSTANT) {
+    VectorPtr strings, delims, limits;
+    const vector_size_t numRows = input.size();
+
+    // Functors to create flat vectors, used as is and for lazy vector.
+    auto funcCreateFlatStrings = [&](RowSet /*rows*/) {
+      return makeFlatVector<StringView>(
+          numRows, [&](vector_size_t row) { return StringView{input[row]}; });
+    };
+
+    auto funcCreateFlatDelims = [&](RowSet /*rows*/) {
+      return makeFlatVector<StringView>(
+          numRows, [&](vector_size_t row) { return StringView{delim}; });
+    };
+
+    auto funcCreateFlatLimits = [&](RowSet /*rows*/) {
+      return makeFlatVector<int32_t>(
+          numRows, [&](vector_size_t row) { return limit.value(); });
+    };
+
+    auto funcReverseIndices = [&](vector_size_t row) {
+      return numRows - 1 - row;
+    };
+
+    // Generate strings vector
+    if (isFlat(encodingStrings)) {
+      strings = funcCreateFlatStrings({});
+    } else if (isConstant(encodingStrings)) {
+      strings =
+          BaseVector::wrapInConstant(numRows, 0, funcCreateFlatStrings({}));
+    } else if (isLazy(encodingStrings)) {
+      strings = std::make_shared<LazyVector>(
+          execCtx_.pool(),
+          CppToType<StringView>::create(),
+          numRows,
+          std::make_unique<SimpleVectorLoader>(funcCreateFlatStrings));
+    } else if (isDictionary(encodingStrings)) {
```

Comment:
Do we need to test the lazy and dictionary vector? This tests the framework, not the function itself. @rui-mo @PHILO-HE

Reply:
that's because we have fast path for different encoding here https://github.com/gayangya/velox/blob/gayangya/split_refactor/velox/functions/sparksql/Split.cpp#L64-L66 , otherwise we don't need this part tests.